handlers/dimensions.go

```diff
@@ -1,27 +1,114 @@
 	"github.com/ONSdigital/dp-frontend-filter-flex-dataset/mapper"
 	"github.com/ONSdigital/dp-net/v2/handlers"
+	"github.com/ONSdigital/log.go/v2/log"
 	"github.com/gorilla/mux"
 )
 
 // DimensionsSelector Handler
-func DimensionsSelector(rc RenderClient) http.HandlerFunc {
+func DimensionsSelector(rc RenderClient, fc FilterClient, dc DimensionClient) http.HandlerFunc {
 	return handlers.ControllerHandler(func(w http.ResponseWriter, req *http.Request, lang, collectionID, accessToken string) {
-		dimensionsSelector(w, req, rc, lang)
+		dimensionsSelector(w, req, rc, fc, dc, collectionID, accessToken, lang)
 	})
 }
 
-func dimensionsSelector(w http.ResponseWriter, req *http.Request, rc RenderClient, lang string) {
+func dimensionsSelector(w http.ResponseWriter, req *http.Request, rc RenderClient, fc FilterClient, dc DimensionClient, collectionID, accessToken, lang string) {
+	ctx := req.Context()
 	vars := mux.Vars(req)
+	filterID := vars["filterID"]
+	nameParam := vars["name"]
+
+	logData := log.Data{
+		"filter_id":  filterID,
+		"name_param": nameParam,
+	}
+
+	currentFilter, _, err := fc.GetJobState(ctx, accessToken, "", "", collectionID, filterID)
+	if err != nil {
+		log.Error(ctx, "failed to get job state", err, logData)
+		setStatusCode(req, w, err)
+		return
+	}
+
+	logData["population_type"] = currentFilter.PopulationType
+
+	filterDimension, err := findDimension(nameParam, currentFilter.Dimensions)
+	if err != nil {
+		log.Error(ctx, "failed to find dimension in filter", err, logData)
+		setStatusCode(req, w, err)
+		return
+	}
+
+	logData["filter_name"] = filterDimension.Name
 
-	// TODO: Get name from endpoint
-	dimension := vars["name"]
 	basePage := rc.NewBasePageModel()
-	m := mapper.CreateSelector(req, basePage, dimension, lang)
 
-	rc.BuildPage(w, m, "selector")
+	if !isAreaType(filterDimension) {
+		selector := mapper.CreateSelector(req, basePage, filterDimension.Name, lang)
+		rc.BuildPage(w, selector, "selector")
+		return
+	}
+
+	areaTypes, err := dc.GetAreaTypes(ctx, accessToken, "", currentFilter.PopulationType)
+	if err != nil {
+		log.Error(ctx, "failed to get geography dimensions", err, logData)
+		setStatusCode(req, w, err)
+		return
+	}
+
+	selector := mapper.CreateAreaTypeSelector(req, basePage, lang, areaTypes.AreaTypes, nameParam)
+	rc.BuildPage(w, selector, "selector")
+}
+
+// isAreaType determines if the current dimension is an area type
+func isAreaType(dimension *filter.ModelDimension) bool {
+	if dimension == nil {
+		return false
+	}
+
+	if dimension.IsAreaType == nil {
+		return false
+	}
+
+	return *dimension.IsAreaType
+}
+
+// findDimension attempts to find a dimension based on the dimension value in a URL param.
+func findDimension(selectionParam string, dimensions []filter.ModelDimension) (*filter.ModelDimension, error) {
+	// Params are matched by name, and therefore may contain spaces/escaped punctuation.
+	selection, err := url.QueryUnescape(selectionParam)
```

Comment:
This is a little awkward, but (at least for me, locally) imported dimensions names are human-readable labels, rather than IDs. I _think_ this might be a consequence of [this](https://github.com/ONSdigital/dp-import-cantabular-dimension-options/pull/40), but not 100% sure.

This results in URL's like:
`/datasets/example/editions/2021/versions/3/dimensions/Number of siblings (3 mappings)/options`

Rather than:
`/datasets/example/editions/2021/versions/3/dimensions/siblings_3/options`

@ChrysmOre Not sure if this is just due to my dataset/import, but I presume we want the ID in the URL to avoid ambiguity and awkwardness having to encode/decode to match them up. If we get a frontend filter like this:

`filters/04708d2c-d507-48b2-9243-bec009e0597d/dimensions/number+of+siblings+%283+mappings%29`

We can't immediately use the API client to fetch the dimension by name etc.

Reply:
Yes I am aware of this issue.
The same logic applies when add a dimension to a filter in a ticket where I paired with @goofballLogic  We look up the name not the id.

Reply:
Yes, I think the model needs to be updated to return the human friendly name and the id and then the other services can be updated to use the appropriate property